src/main/java/net/fortuna/ical4j/model/DateTime.java

```diff
@@ -529,7 +529,7 @@ private DateFormatCache(DateFormat dateFormat) {
 			this.templateFormat = dateFormat;
 		}
 
-		public DateFormat get() {
+		public synchronized DateFormat get() {
```

Comment:
@ztravis 
synchronized scope, can be more smaller, like below.

`private final Object getDateFormatLock = new Object();

public DateFormat get() {
	DateFormat dateFormat;
	synchronized (getDateFormatLock) {
		dateFormat = (DateFormat) threadMap.get(Thread.currentThread());
	}
	...
	return dateFormat;
}`